src/server/api/organization.js

```diff
@@ -46,32 +46,54 @@ export const resolvers = {
       "I'm opting you out of texts immediately. Have a great day.",
     textingHoursStart: organization => organization.texting_hours_start,
     textingHoursEnd: organization => organization.texting_hours_end,
-    twilioAccountSid: organization =>
-      organization.features.indexOf("TWILIO_ACCOUNT_SID") !== -1
-        ? JSON.parse(organization.features).TWILIO_ACCOUNT_SID
-        : null,
-    twilioAuthToken: organization =>
-      organization.features.indexOf("TWILIO_AUTH_TOKEN_ENCRYPTED") !== -1
-        ? JSON.parse(organization.features).TWILIO_AUTH_TOKEN_ENCRYPTED
-        : null,
-    twilioMessageServiceSid: organization =>
-      organization.features.indexOf("TWILIO_MESSAGE_SERVICE_SID") !== -1
-        ? JSON.parse(organization.features).TWILIO_MESSAGE_SERVICE_SID
-        : null,
+    twilioAccountSid: async (organization, _, { user }) => {
+      try {
+        await accessRequired(user, organization.id, "OWNER");
+        return organization.features.indexOf("TWILIO_ACCOUNT_SID") !== -1
+          ? JSON.parse(organization.features).TWILIO_ACCOUNT_SID
+          : null;
+      } catch (err) {
+        return null;
+      }
+    },
+    twilioAuthToken: async (organization, _, { user }) => {
+      try {
+        await accessRequired(user, organization.id, "OWNER");
+        return organization.features.indexOf("TWILIO_AUTH_TOKEN_ENCRYPTED") !==
```

Comment:
One slight issue here. If you clear out the auth token in the settings form, it gets saved in the json as an empty string — which means `organization.features.indexOf("TWILIO_AUTH_TOKEN_ENCRYPTED") !== -1` will always be true after it's been set once. Not that big a deal, but confusing in the UI because it will still show <Encrypted> even though it's actually empty.

Now that I think about it, is there any reason the `indexOf` is needed? Couldn't it just be:
`return JSON.parse(organization.features).TWILIO_AUTH_TOKEN_ENCRYPTED ? "<Encrypted"> : null;`

Reply:
I'll switch it to that.